mayastor/src/core/descriptor.rs

```diff
@@ -109,3 +166,41 @@ impl Debug for Descriptor {
         )
     }
 }
+
+extern "C" fn spdk_range_cb(
```

Comment:
I think we can use the cb handlers for this in the ffihelper.rs ?

Reply:
The problem with the cb handlers was that they create a new oneshot sender. As mentioned above, the unlock expects the same parameters as the lock, so this would cause the unlock to return an error.

Reply:
I think here I might indeed not fully get it but:

the callback here has a fixed address with the code, it by definition always lives on the same address. So it seems to be the case that the void *ctx needs to be point to the same thing, what is _in_ the thing is not something that it cares about, right?